handler/routers/routes.go

```diff
@@ -11,12 +12,14 @@ type Router interface {
 }
 
 type router struct {
+	redisClient producers.RedisClient
+
 	userRouter UserRouter
 }
 
-func NewRouter(controllers controllers.Controllers) Router {
-	userRouter := NewUserRouter(controllers.UserController())
-	return &router{userRouter: userRouter}
+func NewRouter(controllers controllers.Controllers, redisClient producers.RedisClient) Router {
```

Comment:
Where is the rate limiter going to be used? It's just injected here!

Reply:
It is the storage for the rate limiter, and it will be used for routes that require the rate limiter middleware.